server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/XChangeExchange.java

```diff
@@ -64,30 +64,36 @@
 
     private String preferredFiatCurrency;
     private static final long cacheRefreshSeconds = 30;
-    private static final Cache<String, BigDecimal> rateCache = createCache();
+    private static final Cache<String, BigDecimal> rateCache = CacheBuilder
+        .newBuilder()
+        .expireAfterWrite(cacheRefreshSeconds, TimeUnit.SECONDS)
+        .build();
 
-    private static Cache<String, BigDecimal> createCache() {
-        return CacheBuilder
-                .newBuilder()
-                .expireAfterWrite(cacheRefreshSeconds, TimeUnit.SECONDS)
-                .build();
-    }
+    private Exchange exchange = null;
 
-    protected final Exchange exchange;
     private final String name;
     protected final Logger log;
     private final RateLimiter rateLimiter;
+    private final ExchangeSpecification exchangeSpecification;
 
     public XChangeExchange(ExchangeSpecification specification, String preferredFiatCurrency) {
-        exchange = ExchangeFactory.INSTANCE.createExchange(specification);
-        String exchangeName = exchange.getExchangeSpecification().getExchangeName();
-        String sslUri = exchange.getExchangeSpecification().getSslUri();
+        exchangeSpecification = specification;
+        String exchangeName = exchangeSpecification.getExchangeName();
+        String sslUri = exchangeSpecification.getSslUri();
         name = exchangeName + " (" + sslUri + ")"; // just for logging, do not setExchangeName() as it's used to load configuration json internally
         log = LoggerFactory.getLogger("batm.master.exchange." + exchangeName);
         rateLimiter = RateLimiter.create(getAllowedCallsPerSecond());
         this.preferredFiatCurrency = preferredFiatCurrency;
     }
 
+    protected Exchange getExchange() {
```

Comment:
make this method synchronized. just in case it gets called twice in same time.

Reply:
done